contents/code/main.js

```diff
@@ -25,12 +30,16 @@ function isManaged(window) {
 function windowAdded(window) {
 	tryManage(window);
 	if (isManaged(window)) {
-		if (isMaximized(window)) {
+		if (shouldHideTitle(window)) {
 			window.noBorder = true;
 		}
-		window.maximizedChanged.connect(() => {
-			window.noBorder = isMaximized(window);
-		});
+		function statusChanged() {
+			window.noBorder = shouldHideTitle(window);
+		}
+		window.maximizedChanged.connect(statusChanged);
+		if (shouldHideTiled) {
+			window.tileChanged.connect(statusChanged);
+		}
```

Comment:
```suggestion
		window.tileChanged.connect(statusChanged);
```
No need to check this here, since it's also checked in `shouldHideTitle()`. Won't make a difference at the moment, but hopefully in the future config reloading will work properly in Kwin, and then we'll want to attach it always in case the setting changes.

Reply:
Thanks! Fixed in [ca73d27](https://github.com/fin444/truely-maximized/pull/6/commits/ca73d27a1a326d2c0058073f7012ff0f47f1b337).